OutSystems/outsystems-datagrid-reactive
# ROU-3221: Fixing some undo/redo issues on parent dropdown columns

This PR fixes some issues with undo/redo on parent dropdown columns. 

There is still an existing issue that was not solved. If you delete multiple cells from a parent dropdown and press Ctrl + Z, only the first child column cell is reverted
![rou3221-5](https://user-images.githubusercontent.com/3113318/162925489-c294c8bd-ee75-4200-9452-81d0d7f9c951.gif)
.

### What was happening
* whenever we delete multiple cells from a column that has validation and undo the changes, the validation marks are not removed.
![rou3221](https://user-images.githubusercontent.com/3113318/162921273-7e913de9-d5a1-4d41-8cc7-68598fcd43a0.gif)
* The child column was not being reverted after undo/redo.
![rou3221-3](https://user-images.githubusercontent.com/3113318/162921282-1f0a3767-a7cb-4cf6-900e-48f48d639f0e.gif)


### What was done
* Copied Wijmo's GridEditAction class into our codebase, allowing us to add the parent/child cell change onto the stack.
* On _parentCellValueChangeHandler, added new GridEditAction as a child of the existing undoable action from parent column.
* Adjusted undoActionHandler/redoActionHandler to handle child actions as well.

### Test Steps
1. Delete a cell from Employee Department.
2. Press ctrl + Z
Expected: Cells from Department and first name should go back to original value without dirty/invalid mark.

1. Delete a cell from Employee Department.
2. Press ctrl + Z
3. Press ctrl + Y
Expected: Cells from Department and first name should be blank with dirty/invalid mark.

1. Select multiple cells from Employee Department
2. Delete them.
4. Press ctrl + Z
Expected: Cells from Department should go back to original value without dirty/invalid mark.

1. Select multiple cells from Employee Department
2. Delete them.
3. Press ctrl + Z
4. Press ctrl + Y
Expected: Cells from Department should be blank with dirty/invalid mark.
